### PR TITLE
Merge release/12.6-rc-1 into trunk (`beta`)

### PR DESCRIPTION
**Pull request by ParaskP7**
- [x] `build.gradle` file on `WooCommerce` module updated  with version bump. (d81f1a9ae397413796ede88211316a2a7e801c8e)
- [x] `release notes` related files updated. (813e5f327a26a352079b74f0fdf24522b20fbe80 + b7df69816d621929faba5c7e5dc2c6227a281b19 + 7fbf93b9897f641d7e9e30f115bc16bc01355d80 + 3e0678a3355adbf762928b2a03c60664b4118592)
- [x] `new strings` frozen/copied into `fastlane/resources/values/strings.xml`. (432a349d769fad030d9bab09b4f879732c172400)
- [x] `12.6-rc-1` WCAndroid beta submitted to Google for review (`Full rollout`).

EXTRAS:
- [x] `Gemfile` + `Gemfile.lock` release toolkit file update as recommended by automation, plus some manual `wpmreleasetoolkit ` + `bundle install`, followed by another manual `rmagick` update, plust another manual `bundle update`. (d65a813ab2420ed60a2aec9ae1ec496c29d1978d + c3f2d32c39b7239bf3997b294cdd96c9628f4a22 + ff8929d98b3a2484dfa346431791daaf9f0987f4 + 32fa55d9317b638d93a2511e1e9b6e65f16f73fa)
- [x] `fluxc` version bump. (57fd15fde270deec3bebe2c3e0c967af52466361)
- [x] #8477 (b4890f9ab3cb81882ab53452e19cbf1600e1fd75 + 21487d846c92cb2c7812f8da40e90fb919c35d2f + 90c13769a99a80361786662ddf0ca40e511a48f3 + 2dab7b770f9f68bc15acb6e7cc701924b836f919 + 0661fb071460f48e804c0c2daa5c8a7b514e21b8 + f15bdbb27957a763ec914d2d0065fd4f286efbf8 + 89b64efd248ee7cd71c918c91be3914be9290219)
- [x] #8481 (e9b7683fc6748b5df25a596206a06d18ab75efaa + 40797fca69c0b1e60b9744e1c15760b580222289 + abc3bcc5aa77b8dfa1d8fb4e2c09e5b492f340d8)
    - FYI: Note the fact that this PR is not included in `12.6-rc-1` beta release. Instead, it will be included on the next `12.6-rc-2` beta release ([here](https://github.com/woocommerce/woocommerce-android/pull/8500)).

### CODE REVIEW

**Comment by ParaskP7** [comment]
👋 @AliSoftware @jkmassel !

> Gemfile + Gemfile.lock release toolkit file update as recommended by automation, plus some manual wpmreleasetoolkit  + bundle install, followed by another manual rmagick update. (https://github.com/woocommerce/woocommerce-android/commit/d65a813ab2420ed60a2aec9ae1ec496c29d1978d + https://github.com/woocommerce/woocommerce-android/commit/c3f2d32c39b7239bf3997b294cdd96c9628f4a22 + https://github.com/woocommerce/woocommerce-android/commit/ff8929d98b3a2484dfa346431791daaf9f0987f4)

About this checklist item above, and in order for me to achieve this update, I followed Jeremy's steps on a similar update he did on WPAndroid (see [PR](https://github.com/wordpress-mobile/WordPress-Android/pull/17992)). As such I followed the below WPAndroid such commits to achieve the same update on WCAndroid:
- [Update Fastlane](https://github.com/wordpress-mobile/WordPress-Android/pull/17992/commits/5f9264e02b63979fd8383b90b700dcf7f3ad7a2e)
- [Update Release Toolkit](https://github.com/wordpress-mobile/WordPress-Android/pull/17992/commits/d9e5918b8d2da3d0614a68bd39fcd73762faa369)
- [Update nokogiri and rmagick](https://github.com/wordpress-mobile/WordPress-Android/pull/17992/commits/2604e33a7526d7d7da6740872a0f02f19498e40f)

I would love for you to confirm that what I did wasn't terribly wrong... many thanks for all your help! 🙇 🙏

**Comment by AliSoftware** [comment]
@ParaskP7 given that was a major version bump of the release toolkit (aka breaking change), did you follow the instructions that fastlane printed, and especially **check that any breaking changes introduced by the toolkit in this major 7.0.0 version bump wouldn't require an adjustment in your `Fastfile` on method calls?**

Haven't looked in details in the diff of your PR (still busy with my Tumblr duties) but that's the most important thing to check when doing a major version bump of any of our dependencies, so just wanted to check you checked those. Maybe there will be nothing to change if the breaking change it introduced impacted an action you never use or something, but worth the double-check anyway

<details><summary>Message printed in the logs when you run fastlane while still using an old toolkit version and it detects a new major version is available</summary>

```
---------------------------------------
--- Step: check_for_toolkit_updates ---
---------------------------------------
Currently using release toolkit version 6.3.0.
Checking for updates now...
There is a newest version 7.0.0 of the release toolkit!
The latest version available (7.0.0) introduces breaking changes compared to the 6.3.0 you are currently using.

 - To update to 7.0.0, first edit your `Pluginfile` to use '~> 7.0', run `bundle update`,
   then be sure to make all the necessary changes to your `Fastfile` (see the toolkit's CHANGELOG)
   to take those breaking changes into account.

 - If you are not ready to make the major version bump, you can still try to update to the latest compatible,
   non-breaking version by running `bundle update` now. This will not update to the latest 7.0.0, but
   might still update to a newer version compatible with '~> 6.3' if one exists; which is still valuable
   to at least get bugfixes, until you are ready to jump to the next major version later.
```

**Comment by ParaskP7** [comment]
👋 @AliSoftware !

> @ParaskP7 given that was a major version bump of the release toolkit (aka breaking change), did you follow the instructions that fastlane printed...

For some reason I actually didn't read the instruction as I usually just depend on the the `(y/n)` prompt to do this automatic update for me... 😞 

What I did however was the below:
1. I actually first clicked `y` when asked `Do you want to run bundle update now?`. This created the first change, which I [committed](https://github.com/woocommerce/woocommerce-android/commit/d65a813ab2420ed60a2aec9ae1ec496c29d1978d).
2. Then, I indeed updated the `Pluginfile` to use `~> 7.0`, but instead of running `bundle update` I run `bundle install`. With that I created the second change, which I [commited](https://github.com/woocommerce/woocommerce-android/commit/c3f2d32c39b7239bf3997b294cdd96c9628f4a22).
3. Finally, I went and manually edited the `rmagick` to point to `4.3.0`. The `nokogiri` was already pointing to `1.14.2`, thus I didn't have to do anything there. This was my third change, which I [committed](https://github.com/woocommerce/woocommerce-android/commit/ff8929d98b3a2484dfa346431791daaf9f0987f4).

I now went back to the `release/12.6` branch and issued a `bundle update`. This created an extra diff for `Gemfile.lock`, should I commit that change as well and push? 🤔 

<details>
    <summary>Gemfile.lock Diff (19)</summary>

```
CFPropertyList (3.0.5) -> CFPropertyList (3.0.6)
...
aws-partitions (1.649.0) -> aws-partitions (1.720.0)
aws-sdk-core (3.164.0) -> aws-sdk-core (3.170.0)
  ...
  aws-partitions (~> 1, >= 1.525.0) -> aws-partitions (~> 1, >= 1.651.0)
  aws-sigv4 (~> 1.1) -> aws-sigv4 (~> 1.5)
  ...
...
aws-sdk-kms (1.58.0) -> aws-sdk-kms (1.63.0)
  aws-sdk-core (~> 3, >= 3.127.0) -> aws-sdk-core (~> 3, >= 3.165.0)
  ...
...  
aws-sdk-s3 (1.116.0) -> aws-sdk-s3 (1.119.1)
  aws-sdk-core (~> 3, >= 3.127.0) -> aws-sdk-core (~> 3, >= 3.165.0)
  ...
...
excon (0.93.1) ->  excon (0.99.0)
...
fastlane (2.210.1) -> fastlane (2.212.1)
...
google-apis-androidpublisher_v3 (0.29.0) -> google-apis-androidpublisher_v3 (0.34.0)
  google-apis-core (>= 0.9.0, < 2.a) -> google-apis-core (>= 0.9.1, < 2.a)
...
google-apis-playcustomapp_v1 (0.12.0) -> google-apis-playcustomapp_v1 (0.13.0)
  google-apis-core (>= 0.9.1, < 2.a) -> google-apis-core (>= 0.11.0, < 2.a)
...
jmespath (1.6.1) -> jmespath (1.6.2)
json (2.6.2) -> json (2.6.3)
...
mini_magick (4.11.0) -> mini_magick (4.12.0)
...
simctl (1.6.8) -> simctl (1.6.10)
...
```

</details>

-----

Also, what would have been the correct process for this kind of updates? Would the below be the correct one:

1. Clicke `y` when asked `Do you want to run bundle update now`.
2. Update the `Pluginfile` to use `~> 7.0`, then run `bundle update`.

> ...and especially check that any breaking changes introduced by the toolkit in this major 7.0.0 version bump wouldn't require an adjustment in your Fastfile on method calls?

Yes, I did that (checked [7.0.0](https://github.com/wordpress-mobile/release-toolkit/releases/tag/7.0.0)) but I also remembered from discussions you had that the breaking change was only iOS related, so I wasn't alerted too much on that and the fact that I needed to adjust the `Fastfile`. Thus, I went ahead and issued this update, seeing also that Jeremy did the same successfully on WPAndroid as well the other day.

> Haven't looked in details in the diff of your PR (still busy with my Tumblr duties) but that's the most important thing to check when doing a major version bump of any of our dependencies, so just wanted to check you checked those. Maybe there will be nothing to change if the breaking change it introduced impacted an action you never use or something, but worth the double-check anyway

👍

**Comment by AliSoftware** [comment]
> What I did however was the below:
> 
> 1. I actually first clicked `y` when asked `Do you want to run bundle update now?`. This created the first change, which I [committed](https://github.com/woocommerce/woocommerce-android/commit/d65a813ab2420ed60a2aec9ae1ec496c29d1978d).
> 2. Then, I indeed update the `Pluginfile` to use `~> 7.0`, but instead of running `bundle update` I run `bundle install`. With that I created the second change, which I [commited](https://github.com/woocommerce/woocommerce-android/commit/c3f2d32c39b7239bf3997b294cdd96c9628f4a22).
> 3. Finally, I went and manually edited the `rmagick` to point to `4.3.0`. The `nokogiri` was already pointing to `1.14.2`, thus I didn't have to do anything there. This was my third change, which I [committed](https://github.com/woocommerce/woocommerce-android/commit/ff8929d98b3a2484dfa346431791daaf9f0987f4).

This works, but doing (1) was a bit useless if you already planned to do (2) right after it.

Basically by doing your step 1 then your step 2, you first updated to the latest minor version that existed before `7.0.0`… and then updated to `7.0.0`. You could have updated to `7.0.0` directly in one step.

If you actually read the instructions printed by fastlane when it detects a new version (see [the spoiler/collapsed block at the end of my previous comment](https://github.com/woocommerce/woocommerce-android/pull/8478#issuecomment-1453821237)), those indicates that it's typically an either/or situation:

 - Either you're ready to do the major version change and adopt the breaking changes that go with it.
    - In which case, you can update the `Pluginfile` to use `~> 7.0` then run `bundle install` to update the toolkit to that major version (i.e. your step 2 above)
    - But you will also have to make sure to also apply any necessary changes to your call sites to account for the breaking changes that this major version bump introduced.
       - Sometimes this implies having to change a lot of things to adjust to the new API
       - Sometimes (more common in our cases, tbh) it's only going to be a smaller or limited change (but you still have to be sure to do it so that things won't crash next time you run your lanes, of course)
       - Sometimes there's no change at all (like here, because the only breaking changes that `7.0.0` introduced was on an action for iOS, but this is an Android repo so this is not an action that your WCAndroid's `Fastfile` uses at all in the first place).
       - So you have to look at the release-toolkit's CHANGELOG to estimate if the transition will be easy or will need some time to apply (depending on how the breaking changes affect your use case) and decide appropriately if it's the right time to update.
 - Or, if you think that the breaking changes that comes with that major version bump will require a lot more work than you have time for today, you can decide to not update the release-toolkit to the latest major version just yet, and postpone that major version bump for later.
    - In that case, you can either touch nothing and keep the version you're having…
    - Or, even if you don't want to update to the next major version, you maybe still want to try to update to the latest **minor** version bump that came prior to the next major version and breaking change. i.e. your step 1 above.
    - So for example say you're still on version `1.2.3` and next major version is `2.0.0`, you're not ready to update to `2.0.0` yet because it will require a lot of changes to your `Fastfile` to adapt, but maybe there were still some minor versions and the latest `1.x.x` version is e.g. `1.3.6` and you want to update to this non-breaking one for now before you find more time to do the bigger work of bumping to the breaking changes of `2.0.0`.


> I now went back to the `release/12.6` branch and issued a `bundle update`. This created an extra diff for `Gemfile.lock`, should I commit that change as well and push? 🤔

TL;DR: Yes. You should always commit changes to a lockfile (like `Gemfile.lock`).

In most if not all dependency managers (bundler, cocoapods, npm, yarn…), the whole point of a lockfile is to lock the versions on which the dependencies were last resolved to ensure that anyone cloning the repo and thus having the same lockfile will get the exact same dependencies, on the exact same versions as the one you used the last time you updated them, when running the `install` command (`bundle install`).

What `bundle install` does is it guarantees that every developer will have the same version of the dependencies installed on their machine when running `bundle install`. It does that by installing exactly what's in the lockfile (`Gemfile.lock`), without trying to re-resolve the dependencies and risking to update to a different version than the one used last time your app was run on a different machine.

By contrast, `bundle update` asks Bundler to **update all your gems to the latest versions**, regardless of the version they were previously locked on in the `Gemfile.lock`. So that means it will re-resolve the dependencies from scratch, ignoring what's in the `Gemfile.lock` lockfile, potentially finding more recent versions since last time it tried (if new versions of certain gems were published by their authors since). Once it has resolved the dependency graph to find all the latest versions that fits the version restrictions you indicated in your `Gemfile` (the things like `~> 1.3`), it will write the resolved versions of all the gems in the `Gemfile.lock`, concretely locking their version to those specific numbers (that's why it's called a lockfile, after all)

The only case in which `bundle install` will actually potentially update one of the gem instead of installing the version that is set in the `Gemfile.lock` is if the locked version for that gem (e.g. `6.3.0`) does not match the restrictions found in the `Gemfile`—which  can happen if you edited your `Gemfile` to change `~>6.0` to `~>7.0` for example). In that case, it will re-resolve that dependency (try to find a version that matches the new `~>7.0` constraint), but only that one (and its transitive dependencies). Similarly, if you edited your `Gemfile` to add or remove new gems, `bundle install` will resolve those and add/remove them from the `Gemfile.lock` accordingly. So basically, `bundle install` will re-resolve only things you changed in your `Gemfile` since last time to account for the new dependencies list and constraints you added/removed, but for anything else it will not touch their version and will continue to use the version that was locked.


All this is very important concepts to understand as an Apps Infra engineer, as it applies to every dependency manager out there (`bundler`, `cocoapods`, `npm`, `yarn`, `composer`, … they all work the same way, with a lockfile and with install vs update commands that behave the same as I described above for `bundle`), and those are key concepts to do proper Semantic Versioning in our tooling. You can read [this guide I wrote for `pod install`/`pod update` a while ago](https://guides.cocoapods.org/using/pod-install-vs-update.html), which is for `cocoapods`, but the principles and concepts are the same as here for `bundler`.

_The only main widely-used dependency manager that I know of that does not respect those principles like `bundler`/`cocoapods`/`npm`/`yarn`/… do… is gradle. Gradle has apparently no lockfile, and is very lose (at least by default) when it comes to respecting semver… which probably doesn't help teaching you the good practices wrt to semver and dependency management if gradle has been your only experience at doing dependency management for years…_

> Also, what would have been the correct process for this kind of updates?

To… follow the either/or instructions provided by fastlane? 😅 

> Would the below be the correct one:
> 
> 1. Clicked `y` when asked `Do you want to run bundle update now.
> 2. Update the `Pluginfile` to use `~> 7.0`, then run `bundle update`.

As stated above, no need to do 1 if you plan to 2. No need to update the the last minor version if you plan to update to the next major version immediately afterwards…

So if you are ready to update to the next breaking change that it detected:
1. Update the `Pluginfile` with the appropriate next `~> x.y` value for that next major version
2. Run `bundle install`, to only have bundler re-resolve `release-toolkit` (for which you just changed the version constrants, so it will see that the `6.3.0` version locked in your `Gemfile.lock` doesn't comply anymore and will re-resolve it to match `~>7.0`, but will keep other dependencies locked to their version.
3. Do any manual changes in your `Fastfile`, if any needed, to account for the breaking changes that are announced in the `release-toolkit`'s CHANGELOG (none in the case of migrating from `6.x` to `7.x` on an Android repo because only breaking change was on an `ios_*` action, but that might not always be the case)

Sometimes, we use the occasion of updating the release-toolkit on our repos (steps above) to not _just_ update `release-toolkit` (while keeping other gems locked to their current version…) but instead to also update all the other gems while we're at it. If that's what you want to do and want to use the occasion for that, then you'd run `bundle update` (instead of just `bundle install`) on step 2. This is totally fine as long as it was indeed your intent to do so and use the occasion of updating one gem to update them all while at it. This is what Jeremy did when he consciously decided to update `rmagick` in the example WPAndroid commit you linked to, for example. It was not strictly necessary, but just a good occasion to do so while at it.

**Comment by ParaskP7** [comment]
👋 @AliSoftware and thank you so much for the detailed answer(s), this will help me a lot next time and I'll be sure to study all that next week to understand this process fully. 💎 

> This works, but doing (1) was a bit useless if you already planned to do (2) right after it.

👍 

> Basically by doing your step 1 then your step 2, you first updated to the latest minor version that existed before 7.0.0… and then updated to 7.0.0. You could have updated to 7.0.0 directly in one step.

👍 

> If you actually read the instructions printed by fastlane when it detects a new version (see https://github.com/woocommerce/woocommerce-android/pull/8478#issuecomment-1453821237), those indicates that it's typically an either/or situation:
> Either...
> Or...

👍 

> TL;DR: Yes. You should always commit changes to a lockfile (like Gemfile.lock).

Great, thanks for the confirmation, I just pushed that `bundle update` commit (see [here](https://github.com/woocommerce/woocommerce-android/pull/8478/commits/32fa55d9317b638d93a2511e1e9b6e65f16f73fa)).

> To… follow the either/or instructions provided by fastlane? 😅

👍 

> As stated above, no need to do 1 if you plan to 2. No need to update the the last minor version if you plan to update to the next major version immediately afterwards…

👍 

> So if you are ready to update to the next breaking change that it detected:
> 1. ...
> 2. ...
> 3. ...

👍 

> Sometimes, we use the occasion of updating the release-toolkit on our repos (steps above) to not just update release-toolkit (while keeping other gems locked to their current version…) but instead to also update all the other gems while we're at it.

👍 

> If that's what you want to do and want to use the occasion for that, then you'd run bundle update (instead of just bundle install) on step 2. This is totally fine as long as it was indeed your intent to do so and use the occasion of updating one gem to update them all while at it.

👍

> This is what Jeremy did when he consciously decided to update rmagick in the example WPAndroid commit you linked to, for example. It was not strictly necessary, but just a good occasion to do so while at it.

👍

**Comment by ParaskP7** [comment]
👋 @AliSoftware @jkmassel !

Now that I understood all the above and after me pushing this last [commit](https://github.com/woocommerce/woocommerce-android/pull/8478/commits/32fa55d9317b638d93a2511e1e9b6e65f16f73fa), I believe this PR is ready. 🤞 

I will not admin merge this PR, instead I'll wait for you to review it, verify that everything I did with the `release toolkit` update is indeed okay and then let it merge on approval (or else I'll work on any request changes you ask of me). PS: Also, if need be, feel free to push any updates to the `release toolkit` configuration yourself.

FYI: Overall the 4 release toolkit commits that took place in this PR are the following:
1. [Release: Update release toolkit from 6.3.0 to 7.0.0](https://github.com/woocommerce/woocommerce-android/pull/8478/commits/d65a813ab2420ed60a2aec9ae1ec496c29d1978d)
2. [Release: Update wpmreleasetoolkit to 7.0 and then run bundle install](https://github.com/woocommerce/woocommerce-android/pull/8478/commits/c3f2d32c39b7239bf3997b294cdd96c9628f4a22)
3. [Release: Update rmagick to 4.3.0](https://github.com/woocommerce/woocommerce-android/pull/8478/commits/ff8929d98b3a2484dfa346431791daaf9f0987f4)
4. [Release: Run bundle update after updating wpmreleasetoolkit to 7.0](https://github.com/woocommerce/woocommerce-android/pull/8478/commits/32fa55d9317b638d93a2511e1e9b6e65f16f73fa)

**Comment by ParaskP7** [comment]
👋 @woocommerce/apps-infrastructure !

Blocker (🚫): I added the `status: do not merge` label on this PR, while also adding the `[DO NOT MERGE]` prefix to this PR's title, so that to avoid merging this release PR until me, @ThomazFB and the Woo team in general, figure out how to progress with this [hotfix PR](https://github.com/woocommerce/woocommerce-android/pull/8481) that got merged into `release/12.6` 2 days ago (Slack thread for more details: `C6H8C3G23/p1678088917212259`)

-----

👋 @AliSoftware !

Reminder (⏲️): It would still be great if I get a 👍 comment from you on the `release-toolkit` update I did to `7.0.0`. Else, I'll just revert those 4 commits and return on using the previous `6.3.0` version (at least for now). 🙇

**Review by AliSoftware — APPROVED** [review]

**Comment by ParaskP7** [comment]
👋 @ThomazFB !

FYI: I am going to merge this PR to `trunk` now, but I haven't created an `12.6-rc-2` based on your latest https://github.com/woocommerce/woocommerce-android/pull/8481 update. Thus, please expect this inconsistency in terms of what is merged on `trunk` and what's live in an `RC` already.

This is because I can't create a new beta release unless this `12.6-rc-1` branch is merged into `trunk` and we follow the standard process afterwards.

PS: @hichamboushaba is also considering adding [this](https://github.com/woocommerce/woocommerce-android/pull/8493) to this `12.6-rc-2`. Please coordinate with each other and let me know if you want me to create an `12.6-rc-2` with just [yours](https://github.com/woocommerce/woocommerce-android/pull/8481) or wait a bit longer until both betafixes reach the `release/12.6` branch.
